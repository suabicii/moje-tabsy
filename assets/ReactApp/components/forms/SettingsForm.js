import React, {useState} from "react";

function SettingsForm() {
    const darkModeDefault = JSON.parse(localStorage.getItem('darkMode')) || false;
    const [darkMode, setDarkMode] = useState(darkModeDefault);

    return (
        <div className="card mt-3">
            <div className="card-header text-center">Ustawienia ogólne</div>
            <div className="card-body">
                <form role="form" name="settings_form" onSubmit={e => {
                    e.preventDefault();
                    localStorage.setItem('darkMode', `${darkMode}`);
                }}>
                    <div className="notifier-choice">
                        <label className="form-label">Ciemny motyw:</label>
                        <div className="form-check form-switch">
                            <input type="checkbox" name="messengerCheck" id="dark-mode-switcher"
                                   className="form-check-input" role="switch" data-testid="dark-mode-switcher"
                                   checked={darkMode}
                                   onChange={() => setDarkMode(prevState => !prevState)}/>
                            <label htmlFor="dark-mode-switcher"
                                   className="form-check-label">{darkMode ? 'Włączony' : 'Wyłączony'}</label>
                        </div>
                    </div>
                    <div className="d-grid mt-3">
                        <button className="btn btn-primary" data-testid="settings-submit">
                            Zapisz <i className="fa-solid fa-floppy-disk"></i>
                        </button>
                    </div>
                </form>
            </div>
        </div>
    );
}

export default SettingsForm;