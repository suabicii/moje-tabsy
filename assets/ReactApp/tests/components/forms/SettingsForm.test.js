/**
 * @jest-environment jsdom
 * */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import SettingsForm from "../../../components/forms/SettingsForm";
import {fireEvent, render, screen} from "@testing-library/react";

const renderForm = () => {
    render(<SettingsForm/>);
};

it('should correctly render SettingsForm', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<SettingsForm/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('should add remind time input after increasing reminder amount value', () => {
    renderForm();

    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 2}});

    expect(screen.getAllByTestId('remindTime').length).toBeGreaterThan(1);
});

it('should remove remind time input added before after decreasing reminder amount value', () => {
    renderForm();

    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 2}});
    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 1}});

    expect(screen.getAllByTestId('remindTime').length).toBe(1);
});

it('should add remind time input after increasing more than once reminder amount value', () => {
    renderForm();

    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 2}});
    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 3}});

    expect(screen.getAllByTestId('remindTime').length).toBeGreaterThan(2);
});

it('should remove remind time input after decreasing reminder amount value', () => {
    renderForm();

    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 3}});
    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 2}});

    expect(screen.getAllByTestId('remindTime').length).toBe(2);
});

it('should change remind time input amount to correct value after manually writing any reminder amount value any times', () => {
    renderForm();

    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 8}});
    fireEvent.change(screen.getByTestId('reminderAmount'), {target: {value: 4}});

    expect(screen.getAllByTestId('remindTime').length).toBe(4);
});