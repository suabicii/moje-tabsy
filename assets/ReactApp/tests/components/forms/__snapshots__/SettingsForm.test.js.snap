// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render SettingsForm 1`] = `
<div
  className="card mt-3"
>
  <div
    className="card-header text-center"
  >
    Ustawienia ogólne
  </div>
  <div
    className="card-body"
  >
    <form
      name="settings_form"
      onSubmit={[Function]}
      role="form"
    >
      <div
        className="notifier-choice"
      >
        <label
          className="form-label"
        >
          Ciemny motyw:
        </label>
        <div
          className="form-check form-switch"
        >
          <input
            className="form-check-input"
            data-testid="dark-mode-switcher"
            id="dark-mode-switcher"
            name="messengerCheck"
            onChange={[Function]}
            role="switch"
            type="checkbox"
          />
          <label
            className="form-check-label"
            htmlFor="dark-mode-switcher"
          >
            Wyłączony
          </label>
        </div>
      </div>
      <div
        className="d-grid mt-3"
      >
        <button
          className="btn btn-primary"
          data-testid="settings-submit"
        >
          Zapisz 
          <i
            className="fa-solid fa-floppy-disk"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
